src/Utils.ps1

```diff
@@ -138,6 +152,16 @@ function Add-PoshGitToProfile([switch]$AllHosts, [switch]$Force, [switch]$WhatIf
         return
     }
 
+    # If the profile script exists and is signed, then we should not modify it
+    if (Test-Path -LiteralPath $profilePath) {
+        $sig = Get-AuthenticodeSignature $profilePath
+        if ($sig.Status -eq [System.Management.Automation.SignatureStatus]::Valid) {
```

Comment:
Well that turned out to be a little ugly as there are three values that can represent an unsigned file (NotValid, UnknownError, IncompatibleFileType) and four values that can represent a signed file.  Switching to testing the SignatureType field.